src/net.c

```diff
@@ -1025,14 +1025,15 @@ int sockgets(char *s, int *len)
     if (!(socklist[i].flags & (SOCK_UNUSED | SOCK_TCL | SOCK_BUFFER)) &&
         (socklist[i].handler.sock.inbuf != NULL)) {
       if (!(socklist[i].flags & SOCK_BINARY)) {
-        /* look for \r too cos windows can't follow RFCs */
+        /* IRC messages are always lines of characters terminated with a CR-LF
+         * (Carriage Return - Line Feed) pair.
+         */
         p = strpbrk(socklist[i].handler.sock.inbuf, "\r\n");
         if (p != NULL) {
+          *p++ = 0;
           while (*p == '\n' || *p == '\r')
-            *p++ = 0;
-          if (strlen(socklist[i].handler.sock.inbuf) > 510)
-            socklist[i].handler.sock.inbuf[510] = 0;
-          strcpy(s, socklist[i].handler.sock.inbuf);
+            p++;
+          strlcpy(s, socklist[i].handler.sock.inbuf, 511);
           if (*p) {
             px = nmalloc(strlen(p) + 1);
             strcpy(px, p);
```

Comment:
While you're at it seems we're missing an `nfree` in the `else` clause 3 below this comment (I can't comment on the specific line 1042). Unless I'm mistaken it'd be nice to have this fixed in this PR.

Reply:
gcc and clang with -fsanitize=address didn't find any leak, and i made sure the suspicious block was executed for my test. last times i didn't trust the compilers i was always wrong, but inserting nfree() didn't result in a double free, so this time you have beaten the machine. Hooray Cizzle!